executor/src/witgen/identity_processor.rs

```diff
@@ -5,25 +5,75 @@ use itertools::{Either, Itertools};
 use lazy_static::lazy_static;
 use number::FieldElement;
 
+use crate::witgen::machines::Machine;
+
 use super::{
     affine_expression::AffineExpression,
-    machines::{FixedLookup, KnownMachine, Machine},
+    machines::{FixedLookup, KnownMachine},
     rows::RowPair,
     EvalResult, EvalValue, FixedData, IncompleteCause,
 };
 
+/// A list of mutable references to machines.
+pub struct Machines<'a, 'b, T: FieldElement> {
+    machines: Vec<&'b mut KnownMachine<'a, T>>,
+}
+
+impl<'a, 'b, T: FieldElement> Machines<'a, 'b, T> {
+    /// Applies a given function to the ith machine and all other machines.
+    /// With #515 implemented, this could be removed as there would not be a need to split the
```

Comment:
Can you summarize 515? I don't really like code referencing github issues because they might not be available.